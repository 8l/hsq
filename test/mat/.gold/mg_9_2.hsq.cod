	top:top top sqmain

.  _i:3

_main:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; bp 0
	bp; sp bp
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t1 0
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t2 0

l1:
	t1; _i Z; Z t1 ?+3; Z Z ?+9; Z; _i t1; _i t1
	Z t1 l2
	t1; t2; c1 t1; _i t1; t1 t2
	t2 (-1)

	t1; _i Z; Z t1; Z; dec t1
	_i; t1 Z; Z _i; Z

	Z Z l1
l2:

	_i; c1 Z; Z _i; Z

l3:
	t1; c3 Z; Z t1; Z; _i t1
	Z t1 l4
	_i (-1)

	t1; t2; _i t1; dec t1; t1 t2
	_i; t2 Z; Z _i; Z

	Z Z l3
l4:

	?+8; sp ?+4; t2; 0 t2; inc sp
	?+8; sp ?+4; t1; 0 t1; inc sp
	sp; bp sp
	?+8; sp ?+4; bp; 0 bp; inc sp
	?+8; sp ?+4; ?+7; 0 ?+3; Z Z 0

sqmain:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; ?+2 0 _main; . ?; inc sp

	Z Z (-1)

. c2:0 c1:48 c3:58

. t1:0 t2:0

. inc:-1 Z:0 dec:1 ax:0 bp:0 sp:-sp
