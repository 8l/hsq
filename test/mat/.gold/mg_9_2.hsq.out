3210123456789